### PR TITLE
added a bibtex parser that extracts identifiers of entries in bib-fil…

### PR DESCRIPTION
**Pull request by TwlyY29**
…es and makes them accessible on the sidebar.

https://www.mail-archive.com/devel@lists.geany.org/msg03221.html

### CODE REVIEW

**Comment by codebrainz** [comment]
universal-ctags/ctags#2137

**Comment by Skif-off** [comment]
You must add `filetypes.bibtex` to the `filetypes` list (see `data/Makefile.am`), otherwise `make install` will not copy this file.

If without BibTeX lexer then maybe add `lexer_filetype=LaTeX` in section `[settings]`? In this case Geany will highlight comments and LaTeX keywords. Instead naked text...

**Comment by TwlyY29** [comment]
Thanks - I added the filetypes-file to the relevant Makefile.
I am not sure though if Latex keywords should be highlighted. Maybe it would be better to add a bibtex lexer in the future?

**Comment by Skif-off** [comment]
> I am not sure though if Latex keywords should be highlighted.

I did not check how it works (if it works) :) But I think highlighting commented lines can be useful.

> Maybe it would be better to add a bibtex lexer in the future?

Maybe.

**Comment by TwlyY29** [comment]
Hi,

I added highlighting of comments by adding the above-mentioned lexer to the Bibtex-filetype-settings. Also, I copied the Bibtex-Parser from the ctags-repository which was merged there recently: universal-ctags/ctags#2137 

Best
Mirco

File: src/tagmanager/tm_parser.h [hunk 1]
```diff
@@ -67,6 +67,7 @@ enum
 	TM_PARSER_PHP,
 	TM_PARSER_PYTHON,
 	TM_PARSER_LATEX,
+	TM_PARSER_BIBTEX,
```
[the remark this answers is not in the document]
Comment (TwlyY29) [reply src/tagmanager/tm_parser.h]: I've moved `TM_PARSER_BIBTEX`to the end and updated the ctags/parsers.h accordingly

**Comment by TwlyY29** [comment]
Here comes your reminder :-)

**Pull request merged**